packages/schema/bind/src/bindings/assemblyscript/wasm-as/templates/env-ts.mustache

```diff
@@ -4,11 +4,21 @@ import {
 } from "./QueryEnv";
 
 export let env: QueryEnv | null = null;
+
+export function requireEnv(): QueryEnv {
```

Comment:
This feels like a concern that the wrapper developer shouldn't have, and instead it should expect the client to ensure an environment is provided if one is specified in the schema.

This type of checking is already being done here in the `WasmWeb3Api` class: https://github.com/polywrap/monorepo/blob/daa694fd5d7e77c907307c38a40ca388ab23b5cc/packages/js/client/src/wasm/WasmWeb3Api.ts#L336-L354

I think instead of adding error handling here, we should instead improve the code posted above to handle errors in a more granular way.

NOTE: The client makes the assumption that the `_w3_load_env` function will be present if the wrapper requires an environment. This is the only logical way to detect this requirement IMO.

Reply:
Yep, I agree, and when we load the env, It will take care of the type on `serialize <-> deserialize` steps. This is just a convenient method for the wrapper developer to get the env in concrete type (i.e: QueryEnv or MutationEnv), not the  (QueryEnv | null) because handling this on the wrapper side just makes the code more verbose, not a big deal but clutters the code.

Reply:
> I think instead of adding error handling here, we should instead improve the code posted above to handle errors in a more granular way.

Can you elaborate on this?